Source/levels/themes.cpp

```diff
@@ -43,45 +43,44 @@ size_t themeVar1;
 
 bool TFit_Shrine(int i)
 {
-	int xp = 0;
-	int yp = 0;
+	Point position { 0, 0 };
 	size_t found = 0;
 
 	while (found == 0) {
-		Point testPosition { xp, yp };
-		if (dTransVal[xp][yp] == themes[i].ttval) {
-			if (TileHasAny(dPiece[xp][yp - 1], TileProperties::Trap)
+		Point testPosition { position.x, position.y };
+		if (dTransVal[position.x][position.y] == themes[i].ttval) {
+			if (TileHasAny(position + DisplacementOf { 0, -1 }, TileProperties::Trap)
 			    && IsTileNotSolid(testPosition + Direction::NorthWest)
 			    && IsTileNotSolid(testPosition + Direction::SouthEast)
-			    && dTransVal[xp - 1][yp] == themes[i].ttval
-			    && dTransVal[xp + 1][yp] == themes[i].ttval
+			    && dTransVal[position.x - 1][position.y] == themes[i].ttval
+			    && dTransVal[position.x + 1][position.y] == themes[i].ttval
 			    && !IsObjectAtPosition(testPosition + Direction::North)
 			    && !IsObjectAtPosition(testPosition + Direction::East)) {
 				found = 1;
 			}
 			if (found == 0
-			    && TileHasAny(dPiece[xp - 1][yp], TileProperties::Trap)
+			    && TileHasAny(position + DisplacementOf { -1, 0 }, TileProperties::Trap)

```

Comment:
Might be cleaner using Direction instead of DisplacmentOf. 
```suggestion
			    && TileHasAny(position + Direction::NorthWest, TileProperties::Trap)
```